charts/vector-experimental/templates/daemonset.yaml

```diff
@@ -0,0 +1,23 @@
+{{- if (eq .Values.role "Agent") -}}
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: {{ include "vector.fullname" . }}
+  labels:
+    {{- include "vector.labels" . | nindent 4 }}
+spec:
+  selector:
+    matchLabels:
+      {{- include "vector.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      {{- with .Values.podAnnotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      labels:
+        {{- include "vector.selectorLabels" . | nindent 8 }}
+        vector.dev/exclude: "true"
```

Comment:
Ported directly from the existing chart - I believe generally used to avoid collecting the agent logs if it's _also_ writing what it's collected back to stdout with a `console` sink (infinite loops and all). Definitely could be optional though.

Reply:
It actually has come up once - https://github.com/timberio/helm-charts/issues/6 so there's at least _some_ desire :)

Reply:
Going to punt on this until I have a better solution - perhaps updated defaults. Writing logs to stdout + an agent collecting is 😭